lib/mongo/socket/ssl.rb

```diff
@@ -23,6 +23,8 @@ class Socket
     class SSL < Socket
       include OpenSSL
 
+      ALLOWED_KEY_TYPES = [OpenSSL::PKey::RSA]
```

Comment:
This was a relic from the first implementation which I had left there in case we needed type checking. But since we are not doing type checking on the objects passed in this constant is  not used and therefore I removed it completely